### DISCUSSION TITLE
Feat(intellisense): format long ints in signatures with separators

### DISCUSSION THREAD

**Discussion by NickCrews** [microsoft/pylance-release#6608] [Enhancement]
Thanks for your help. I'm not sure if this is the correct repo to file this request, please point me in the right direction if this is wrong.

Sometimes I have a big int as a default value of a function:

```python
def train_using_labels(
    comparers: Iterable[LevelComparer],
    left: ir.Table,
    right: ir.Table,
    *,
    max_pairs: int = 1_000_000_000,
) -> Weights:
```

Note that I have the _ separators in there to make it readable. When I hover over a usage of that function, the intellisense popup prints it plain, with no separators. This makes it difficult to read:

![Image](https://github.com/user-attachments/assets/5c29a207-0003-410c-85a3-f3be8d785f99)

Can we either:
1. print it exactly as formatted? This seems more difficult, as I'd imagine after parsing we only are passing around the abstract representation.
2. add some sort of formatting logic, perhaps simply the python equivalent of  adding the _ separator, eg `print(f"{1_000_000_000:_}")`

**Comment by rchiodo** [comment]
Thanks for the suggestion. Transferring to discussions for up votes.

**Comment by debonte** [comment]
Seems like a duplicate of https://github.com/microsoft/pylance-release/discussions/4030?

**Reply by NickCrews** [reply]
yeah, seems like dupe enough to me, I'll go chime in there. Thanks!
